internal/common/function/function.go

```diff
@@ -117,28 +120,137 @@ func GenerateVeleroBackupName(namespace, nabName string) string {
 	return veleroBackupName
 }
 
+// UpdateNonAdminPhase updates the phase of a NonAdminBackup object with the provided phase.
+func UpdateNonAdminPhase(ctx context.Context, r client.Client, logger logr.Logger, nab *nacv1alpha1.NonAdminBackup, phase nacv1alpha1.NonAdminBackupPhase) (bool, error) {
+	if nab == nil {
+		return false, errors.New("NonAdminBackup object is nil")
+	}
+
+	// Ensure phase is valid
+	if phase == constant.EmptyString {
+		return false, errors.New("NonAdminBackupPhase cannot be empty")
+	}
+
+	oldPhase := nab.Status.Phase
+	nab.Status.Phase = phase
+
+	logger.V(1).Info(fmt.Sprintf("Setting NonAdminBackup Phase to: %s", phase))
```

Comment:
this `	nab.Status.Phase = phase` does not already set phase?

Reply:
this log seems to much for info level, already logs at the end of function what the new phase is

Reply:
You are right. Moved the nab.Status.Phase below.

Removed that first debug log. (Note V(1) is debug, so not normally visible when controller is running).